ic-canister/tests/canister_b/src/lib.rs

```diff
@@ -55,6 +55,29 @@ impl CanisterB {
             .unwrap();
         canister_call!(canister_a.get_counter(), u32).await.unwrap()
     }
+
+    #[update]
+    #[allow(unused_mut)]
+    #[allow(unused_variables)]
+    async fn call_increment_oneway(&self, value: u32) -> bool {
+        let mut canister_a = CanisterA::from_principal(self.state.borrow().canister_a);
+
+        #[cfg(target_arch = "wasm32")]
+        canister_call_oneway!(canister_a.inc_counter(value), ()).unwrap();
+
+        #[cfg(not(target_arch = "wasm32"))]
+        canister_call_oneway!(canister_a.inc_counter(value), ()).await.unwrap();
```

Comment:
The usage of the macro must be same in both environments. So both these calls must be not `async`.

Reply:
Yes, I'll figure out the mock calls in a non-wasm environment, and change canister_notify to be non-async